### PR TITLE
fix(Core/Player) Fix Rediscovering Certain Flight Paths and Zones After a Logout and Login

### LINKED ISSUE #7737: Certain Flightpaths and Zones Revert back to undiscovered upon user Logout.
### Current Behaviour

Discovered Flight Paths and Zones revert to undiscovered upon user logout.

Current Example

Nutral - Flight Master in Shattrath City.  (Flight Path) (Terokkar Forest)
Ruins of Lordaron (Zone) (Tirisfal Glades)
Narzyn Skybreaker - Flight Master in Agmar's Hammer (Flight Path) (Dragonblight)
Celea Frozenmane - Flight Master in Camp Winterhoof (Flight Path) (Howling Fjord)

### Expected Blizzlike Behaviour

The zones / flightpaths should stay discovered after user logout.

### Source

_No response_

### Steps to reproduce the problem

1. Create a character
2. .tele Shattrath
3. click on nutral and get flight path.
4. logout
5. log back in
6. Green exclamation mark is above Nutral.

### Extra Notes

It appears that there's at least 1 zone or flight master in each zone affected by this.

### AC rev. hash/commit

AzerothCore rev. 728839b84952 2021-09-05 10:02:31 +0000 (master branch) (Win64, Release, Static)

### Operating system

Windows 10 x64

### Custom changes or Modules

1v1arena
ahbot
AutoBalance
CFBG
mod-double-xp-weekend

### PR DESCRIPTION
**Pull request by LannyE**
Certain zones and flight path discoveries are removed on logout and must
be frustratingly rediscovered again

Closes Azerothcore issue #7737

## Changes Proposed:
-  It apparently fixes the conversion of strings to unsigned longs.  Certain zones and flight paths must not convert right so they do not save on a logout.  
-  Got the info from [https://github.com/TrinityCore/TrinityCore/commit/24c173f387044496d581255545ed94c988f3ae3b](url)
and [https://github.com/TrinityCore/TrinityCore/issues/13493](url)
## Issues Addressed:

- Closes #7737

## Tests Performed:

- Went the to flight masters specified in the Issues (Nutral, Bladruc, Celea) and checked to see if flight the flight was saved on logout.
- Learned spells and recipes and generally tried to break stuff.
- Went to zones like thelsamar and 
- Compiles and runs just fine.

## How to Test the Changes:

1. Go to Baldruc in Theramore and learn his flight path.  Logout and log back in and see if it still is learned/discovered
2. Go to Valley of Fangs in Badlands and discover it then logout and log back and see if you rediscover it.  
3.

## Known Issues and TODO List:

Please test this thing as I am not a great programmer. 
- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by acidmanifesto** [comment]
Code style Check reports:
image

C++ 20 check reports:
image

Ubuntu Checks reports:
image

**Review by acidmanifesto — CHANGES_REQUESTED** [review]
Fix errors a reported by the Test Results.

**Review by Kitzunu — CHANGES_REQUESTED** [review]
I don't think atoul is the right way to fix this.

@Winfidonarleyan we have our own better thing to do that right?

**Review by acidmanifesto — APPROVED** [review]

**Comment by Winfidonarleyan** [comment]
> I don't think atoul is the right way to fix this.
> 
> @Winfidonarleyan we have our own better thing to do that right?

Of course `Acore::StringTo`

File: src/common/Common.h [hunk 1]
```diff
@@ -48,6 +48,8 @@
 #define stricmp strcasecmp
 #endif
 
+inline unsigned long atoul(char const* str) { return strtoul(str, nullptr, 10); }
```
Comment (Winfidonarleyan) [inline src/common/Common.h:51]: No need this.
https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/common/Utilities/StringConvert.h

**Comment by Winfidonarleyan** [comment]
I will make a more global PR

**Comment by LannyE** [comment]
Do you still need me to make the requested change or anything else?

**Comment by Winfidonarleyan** [comment]
> Do you still need me to make the requested change or anything else?

Use StringConvert

**Comment by LannyE** [comment]
I am working on the conversion.  Be warned my C++ skills are not that strong.  In my attempt to use `Acore::StringTo` I replaced this code in PlayerDump.cpp line 554:
`race = uint8(atoul(getnth(line, 4).c_str()));`
with
`race = uint8(Acore::StringTo<uint8>(getnth(line, 4).c_str()));`
and I get this error:
`Error	C2440	'<function-style-cast>': cannot convert from 'std::optional<uint8>' to 'uint8'`
I haven't been able to figure it out.  Any guidance on how to fix this would be appreciated.

**Comment by LannyE** [comment]
Nevermind it appears PR #10121 takes care of most of this except the Chat.cpp and PlayerDump.cpp.  I just tested it and that PR does fix the flight paths and the rediscovering zones issue from #7737.  As a side note the tokenizer PR #10121 breaks the Arena 1v1 module.  That module uses the tokenizer code so now you can't compile new versions of Azerothcore with that module present.  

Do you want to close this PR since the issue it was created for is now resolved?

**Comment by Winfidonarleyan** [comment]
> Nevermind it appears PR #10121 takes care of most of this except the Chat.cpp and PlayerDump.cpp. I just tested it and that PR does fix the flight paths and the rediscovering zones issue from #7737. As a side note the tokenizer PR #10121 breaks the Arena 1v1 module. That module uses the tokenizer code so now you can't compile new versions of Azerothcore with that module present.
> 
> Do you want to close this PR since the issue it was created for is now resolved?

if problem resolved - can close. I continue replace `ato**` next time

**Pull request closed without merging**